src/main/msp/msp_serial.c

```diff
@@ -367,25 +367,25 @@ static mspPostProcessFnPtr mspSerialProcessReceivedCommand(mspPort_t *msp, mspPr
     uint8_t outBuf[MSP_PORT_OUTBUF_SIZE];
 
     mspPacket_t reply = {
-        .buf = { .ptr = outBuf, .end = ARRAYEND(outBuf), },
         .cmd = -1,
         .flags = 0,
         .result = 0,
     };
-    uint8_t *outBufHead = reply.buf.ptr;
+    sbufInitialize(&reply.buf, outBuf, ARRAYEND(outBuf));
 
     mspPacket_t command = {
-        .buf = { .ptr = msp->inBuf, .end = msp->inBuf + msp->dataSize, },
         .cmd = msp->cmdMSP,
         .flags = msp->cmdFlags,
         .result = 0,
     };
+    sbufInitialize(&command.buf, msp->inBuf, msp->inBuf + msp->dataSize);
 
     mspPostProcessFnPtr mspPostProcessFn = NULL;
     const mspResult_e status = mspProcessCommandFn(&command, &reply, &mspPostProcessFn);
 
-    if (status != MSP_RESULT_NO_REPLY) {
-        sbufSwitchToReader(&reply.buf, outBufHead); // change streambuf direction
+    // Reply only if we have a reply and it's not truncated by sbuf overflow protection
```

Comment:
I would reply with an MSP error. According to some code in msp.js in the configurator, we could send the `!` character as the direction char (i.e. instead of `>`) to indicate an error. This way the configurator can show an alert to the user and we will be able to diagnose it more easily (e.g. "Error response received to command 42: Buffer was too short for response" or "...: Buffer was to short for request" or "...: Additional unexpected data at the end of the buffer" - you get the idea).

Also, I would also consider an error if the request contained X bytes and the FC consumed just Y, where Y < X.